components/suggestion/command_provider/app_command_parser.ts

```diff
@@ -43,38 +41,77 @@ export type Store = {
     getState: () => GlobalState;
 }
 
+enum ParseState {
+    Start = 0,
+    Command,
+    EndCommand,
+    CommandSeparator,
+    Parameter,
+    ParameterSeparator,
+    Flag1,
+    Flag,
+    FlagValueSeparator,
+    StartValue,
+    NonspaceValue,
+    QuotedValue,
+    TickValue,
+    EndValue,
+}
+
+interface ParsedCommand {
+    text: string;
+    state: ParseState;
+    incomplete: string;
+    incompleteStart: number;
+    binding: AppBinding | null;
+    form: AppForm;
+    field: AppField;
+    values: {[name: string]: string};
+    location: string;
+    error: string;
+}
```

Comment:
This may work to have all other fields be optional when `error` is present

```suggestion
} | {
    error: string;
}
```